<a href="https://colab.research.google.com/github/Shikhashah88/datasci_1_loading/blob/main/Datasets.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**The following code cell below imports pandas**

we need pandas to load our datasets



In [88]:
import pandas as pd



In [89]:
df1 = '/content/DATA/Healthcare-Diabetes.csv'
read_csv1 = pd.read_csv(df1)
read_csv1.head(11)


,Id,Pregnancies,Glucose,BloodPressure,SkinThickness,Insulin,BMI,DiabetesPedigreeFunction,Age,Outcome
0,1,6,148,72,35,0,33.6,0.627,50,1
1,2,1,85,66,29,0,26.6,0.351,31,0
2,3,8,183,64,0,0,23.3,0.672,32,1
3,4,1,89,66,23,94,28.1,0.167,21,0
4,5,0,137,40,35,168,43.1,2.288,33,1
5,6,5,116,74,0,0,25.6,0.201,30,0
6,7,3,78,50,32,88,31.0,0.248,26,1
7,8,10,115,0,0,0,35.3,0.134,29,0
8,9,2,197,70,45,543,30.5,0.158,53,1
9,10,8,125,96,0,0,0.0,0.232,54,1


# Diabetes Prediction Dataset

## Description

The Diabetes Prediction Dataset contains health-related attributes that can be used to explore relationships between health indicators and the likelihood of diabetes. The dataset can be used for data analysis, visualization, feature selection, and predictive modeling.

## Columns / Data Dictionary

| Column | Description |
|---|---|
| **Id** | Unique identifier for each data entry. |
| **Pregnancies** | Number of times pregnant. |
| **Glucose** | Plasma glucose concentration over 2 hours in an oral glucose tolerance test. |
| **BloodPressure** | Diastolic blood pressure (mm Hg). |
| **SkinThickness** | Triceps skinfold thickness (mm). |
| **Insulin** | 2-hour serum insulin (mu U/ml). |
| **BMI** | Body mass index (weight in kg / height in m²). |
| **DiabetesPedigreeFunction** | Diabetes pedigree function, representing a genetic score related to diabetes. |
| **Age** | Age of the individual in years. |
| **Outcome** | Binary classification indicating the presence (1) or absence (0) of diabetes. |

## Purpose of the Dataset

This dataset can be used to explore the relationships between health indicators and diabetes outcomes. It can also be used to develop predictive models and create visualizations to identify patterns associated with diabetes risk.



In [90]:
df2 = '/content/DATA/Heart-Disease-UCI.csv'
read_csv2 = pd.read_csv(df2)
read_csv2.head(11)


,age,sex,cp,trestbps,chol,fbs,restecg,thalch,exang,oldpeak,num
0,63,Male,typical angina,145.0,233.0,True,lv hypertrophy,150.0,False,2.3,0
1,67,Male,asymptomatic,160.0,286.0,False,lv hypertrophy,108.0,True,1.5,2
2,67,Male,asymptomatic,120.0,229.0,False,lv hypertrophy,129.0,True,2.6,1
3,37,Male,non-anginal,130.0,250.0,False,normal,187.0,False,3.5,0
4,41,Female,typical angina,130.0,204.0,False,lv hypertrophy,172.0,False,1.4,0
5,56,Male,typical angina,120.0,236.0,False,normal,178.0,False,0.8,0
6,62,Female,asymptomatic,140.0,268.0,False,lv hypertrophy,160.0,False,3.6,3
7,57,Female,asymptomatic,120.0,354.0,False,normal,163.0,True,0.6,0
8,63,Male,asymptomatic,130.0,254.0,False,lv hypertrophy,147.0,False,1.4,2
9,53,Male,asymptomatic,140.0,203.0,True,lv hypertrophy,155.0,True,3.1,1


# Heart Disease UCI Dataset

## Description

The Heart Disease UCI Dataset is a multivariate dataset containing health and clinical information about patients. It includes 14 commonly used attributes related to heart disease. The dataset can be used to explore relationships between patient characteristics, clinical measurements, and the presence of heart disease.

## Columns / Data Dictionary

| Column | Description |
|---|---|
| **id** | Unique identifier for each patient. |
| **age** | Age of the patient in years. |
| **origin** | Place where the study was conducted. |
| **sex** | Sex of the patient (Male/Female). |
| **cp** | Chest pain type: 1 = typical angina, 2 = atypical angina, 3 = non-anginal, 4 = asymptomatic. |
| **trestbps** | Resting blood pressure in mm Hg on admission to the hospital. |
| **chol** | Serum cholesterol level in mg/dl. |
| **fbs** | Indicates whether fasting blood sugar is greater than 120 mg/dl. |
| **restecg** | Resting electrocardiographic results: normal, ST-T wave abnormality, or left ventricular hypertrophy. |
| **thalach** | Maximum heart rate achieved. |
| **exang** | Exercise-induced angina (True/False). |
| **oldpeak** | ST depression induced by exercise relative to rest. |
| **slope** | Slope of the peak exercise ST segment. |
| **ca** | Number of major vessels (0–3) colored by fluoroscopy. |
| **thal** | Thalassemia result: normal, fixed defect, or reversible defect. |
| **num** | Predicted attribute indicating the presence or severity of heart disease. |

## Purpose of the Dataset

The dataset can be used to investigate relationships between clinical characteristics and heart disease. It can also be used for data visualization, statistical analysis, feature selection, and machine learning models that predict heart disease.


In [91]:
df3 = '/content/DATA/healthcare-dataset-stroke-data.csv'
read_csv3 = pd.read_csv(df3)
read_csv3.head(11)


,id,gender,age,hypertension,heart_disease,ever_married,work_type,Residence_type,avg_glucose_level,bmi,smoking_status,stroke
0,9046,Male,67.0,0,1,Yes,Private,Urban,228.69,36.6,formerly smoked,1
1,51676,Female,61.0,0,0,Yes,Self-employed,Rural,202.21,NaN,never smoked,1
2,31112,Male,80.0,0,1,Yes,Private,Rural,105.92,32.5,never smoked,1
3,60182,Female,49.0,0,0,Yes,Private,Urban,171.23,34.4,smokes,1
4,1665,Female,79.0,1,0,Yes,Self-employed,Rural,174.12,24.0,never smoked,1
5,56669,Male,81.0,0,0,Yes,Private,Urban,186.21,29.0,formerly smoked,1
6,53882,Male,74.0,1,1,Yes,Private,Rural,70.09,27.4,never smoked,1
7,10434,Female,69.0,0,0,No,Private,Urban,94.39,22.8,never smoked,1
8,27419,Female,59.0,0,0,Yes,Private,Rural,76.15,NaN,Unknown,1
9,60491,Female,78.0,0,0,Yes,Private,Urban,58.57,24.2,Unknown,1


# Stroke Prediction Dataset

## Description

According to the World Health Organization (WHO), stroke is the second leading cause of death globally. This dataset is used to predict whether a patient is likely to have a stroke based on health and demographic characteristics such as gender, age, hypertension, heart disease, glucose level, BMI, and smoking status.

## Columns  | Data Dictionary

| Column | Description |
|---|---|
| **id** | Unique identifier for each patient. |
| **gender** | Gender of the patient: Male, Female, or Other. |
| **age** | Age of the patient. |
| **hypertension** | Indicates whether the patient has hypertension: 0 = No, 1 = Yes. |
| **heart_disease** | Indicates whether the patient has heart disease: 0 = No, 1 = Yes. |
| **ever_married** | Indicates whether the patient has ever been married: No or Yes. |
| **work_type** | Type of employment: children, Govt_job, Never_worked, Private, or Self-employed. |
| **Residence_type** | Type of residence: Rural or Urban. |
| **avg_glucose_level** | Average glucose level in the blood. |
| **bmi** | Body mass index. |
| **smoking_status** | Smoking status: formerly smoked, never smoked, smokes, or Unknown. |
| **stroke** | Indicates whether the patient had a stroke: 0 = No, 1 = Yes. |

## Purpose of the Dataset

This dataset can be used to explore the relationships between demographic and health-related factors and the likelihood of stroke. It can also be used for data visualization, statistical analysis, feature selection, and machine learning models for stroke prediction.



In [92]:
df4 = '/content/DATA/clinical_trial_data.xlsx'
read_excel1 = pd.read_excel(df4)
read_excel1.head(11)


,Subject_ID,Treatment_Arm,Baseline_Score,Post_Treatment_Score,Score_Improvement,Adverse_Events_Reported
0,SUBJ-001,Placebo,45,46,1,0
1,SUBJ-002,Low Dose,48,58,10,1
2,SUBJ-003,High Dose,50,72,22,2
3,SUBJ-004,Placebo,42,43,1,0
4,SUBJ-005,High Dose,49,75,26,3
5,SUBJ-006,Low Dose,46,59,13,1
6,SUBJ-007,Placebo,44,45,1,0
7,SUBJ-008,High Dose,52,78,26,2
8,SUBJ-009,Low Dose,47,60,13,0
9,SUBJ-010,Placebo,43,42,-1,1


# Clinical Trial Dataset

## Description

This dataset contains clinical trial information for 25 subjects. It includes information about treatment assignment, clinical scores before and after treatment, score improvement, and reported adverse events.

## Columns  | Data Dictionary

| Column | Description |
|---|---|
| **Subject_ID** | Unique identifier for each clinical trial subject. |
| **Treatment_Arm** | Treatment or intervention group assigned to the subject. |
| **Baseline_Score** | Clinical score recorded before treatment. |
| **Post_Treatment_Score** | Clinical score recorded after treatment. |
| **Score_Improvement** | Change in the subject's clinical score following treatment. |
| **Adverse_Events_Reported** | Indicates whether the subject reported an adverse event during the clinical trial. |

## Purpose of the Dataset

This dataset can be used to evaluate the effectiveness and safety of a clinical treatment by comparing baseline and post-treatment scores and examining reported adverse events. It can also be used for data visualization, statistical analysis, and exploring differences in treatment outcomes.

In [93]:
df5 = '/content/DATA/heart_failure_clinical_records.xlsx'
read_excel2 = pd.read_excel(df5)
read_excel2.head(11)


,Record_ID,Age,Anaemia,Ejection_Fraction_Pct,High_Blood_Pressure,Serum_Creatinine_mg_dL,Serum_Sodium_mEq_L,DEATH_EVENT
0,1,60,0,38,1,1.1,136,0
1,2,55,0,55,0,0.9,140,0
2,3,65,0,30,0,1.3,132,0
3,4,50,1,20,0,1.9,137,1
4,5,70,1,20,1,2.1,130,1
5,6,62,0,40,1,1.0,138,0
6,7,75,1,35,0,1.4,134,1
7,8,68,0,60,1,0.8,142,0
8,9,52,0,38,0,1.2,139,0
9,10,58,1,25,1,1.8,131,1


# Heart Failure Clinical Records Dataset

## Description

This dataset contains clinical records from patients with heart failure. It includes demographic information, medical conditions, laboratory measurements, and follow-up information. The dataset can be used to examine factors associated with heart failure outcomes.

## Columns  | Data Dictionary

| Column | Description |
|---|---|
| **age** | Age of the patient in years. |
| **anaemia** | Indicates whether the patient has anaemia: 0 = No, 1 = Yes. |
| **creatinine_phosphokinase** | Level of creatinine phosphokinase (CPK) enzyme in the blood. |
| **diabetes** | Indicates whether the patient has diabetes: 0 = No, 1 = Yes. |
| **ejection_fraction** | Percentage of blood leaving the heart during each contraction. |
| **high_blood_pressure** | Indicates whether the patient has high blood pressure: 0 = No, 1 = Yes. |
| **platelets** | Platelet count in the blood. |
| **serum_creatinine** | Level of serum creatinine in the blood. |
| **serum_sodium** | Level of serum sodium in the blood. |
| **sex** | Sex of the patient: 0 = Female, 1 = Male. |
| **smoking** | Indicates whether the patient smokes: 0 = No, 1 = Yes. |
| **time** | Follow-up period in days. |
| **DEATH_EVENT** | Indicates whether the patient died during the follow-up period: 0 = No, 1 = Yes. |

## Purpose of the Dataset

This dataset can be used to explore relationships between demographic characteristics, medical conditions, laboratory measurements, and heart failure outcomes. It can also be used for data visualization, statistical analysis, feature selection, and machine learning models to predict patient outcomes.

In [94]:
df6 = '/content/DATA/hospital_readmissions.xlsx'
read_excel3 = pd.read_excel(df5)
read_excel3.head(11)


,Record_ID,Age,Anaemia,Ejection_Fraction_Pct,High_Blood_Pressure,Serum_Creatinine_mg_dL,Serum_Sodium_mEq_L,DEATH_EVENT
0,1,60,0,38,1,1.1,136,0
1,2,55,0,55,0,0.9,140,0
2,3,65,0,30,0,1.3,132,0
3,4,50,1,20,0,1.9,137,1
4,5,70,1,20,1,2.1,130,1
5,6,62,0,40,1,1.0,138,0
6,7,75,1,35,0,1.4,134,1
7,8,68,0,60,1,0.8,142,0
8,9,52,0,38,0,1.2,139,0
9,10,58,1,25,1,1.8,131,1


# Hospital Readmissions Dataset

## Description

This dataset contains hospital admission information for 25 patients. It includes patient age group, admission type, primary diagnosis, length of hospital stay, and whether the patient was readmitted within 30 days.

## Columns  | Data Dictionary

| Column | Description |
|---|---|
| **Patient_ID** | Unique identifier for each patient. |
| **Age_Group** | Age category or group of the patient. |
| **Admission_Type** | Type of hospital admission, such as emergency, urgent, or elective. |
| **Primary_Diagnosis** | Main medical condition or diagnosis associated with the hospital admission. |
| **Length_of_Stay_Days** | Number of days the patient stayed in the hospital. |
| **Readmitted_30Days** | Indicates whether the patient was readmitted to the hospital within 30 days: 0 = No, 1 = Yes. |

## Purpose of the Dataset

This dataset can be used to explore factors associated with hospital readmissions. It can also be used to examine relationships between age group, admission type, primary diagnosis, and length of stay and whether a patient was readmitted within 30 days. The dataset can support data visualization, statistical analysis, and basic machine learning exercises.

In [95]:
df7 = '/content/DATA/all_products_refined.json'
read_json1 = pd.read_json(df7)
read_json1.head(11)


,name,description,content_summary,ingredients,packaging
0,Well People Supernatural Complexion Stick Foun...,"This product bears the EWG Verified® mark, mea...",The specific formulation has been reviewed by ...,[{'name': 'BAMBUSA ARUNDINACEA (BAMBOO) STEM P...,"{'ingredients': 'Caprylic/Capric Triglyceride,..."
1,"Rejuva Minerals Age Defying Liquid Foundation,...","This product bears the EWG Verified® mark, mea...",The specific formulation has been reviewed by ...,[{'name': 'ALOE BARBADENSIS (ALOE VERA) LEAF J...,"{'ingredients': 'Water (Aqua), Caprylic/Capric..."
2,Beautycounter Skin Twin Featherweight Foundati...,"This product bears the EWG Verified® mark, mea...",The specific formulation has been reviewed by ...,"[{'name': 'BEESWAX', 'function': 'emollient, e...","{'ingredients': '', 'directions': '', 'warning..."
3,"Maia's Mineral Galaxy Mineral Foundation, Soft...","This product bears the EWG Verified® mark, mea...",The specific formulation has been reviewed by ...,"[{'name': 'BORON NITRIDE', 'function': 'absorb...","{'ingredients': 'Boron Nitride, Silica Amorpho..."
4,"Well People Narcissist Foundation Sticks, Rich...","This product bears the EWG Verified® mark, mea...",The specific formulation has been reviewed by ...,"[{'name': 'ALARIA ESCULENTA EXTRACT', 'functio...","{'ingredients': 'Oleic Acid*, Ricinus Communis..."
5,"Maia's Mineral Galaxy Mineral Foundation, Rosy...","This product bears the EWG Verified® mark, mea...",The specific formulation has been reviewed by ...,"[{'name': 'BORON NITRIDE', 'function': 'absorb...","{'ingredients': 'Boron Nitride, Silica Amorpho..."
6,"Rejuva Minerals Luminous Creme Foundation, Haw...","This product bears the EWG Verified® mark, mea...",The specific formulation has been reviewed by ...,"[{'name': 'BEESWAX', 'function': 'emollient, e...","{'ingredients': 'Zinc Oxide, Simmondsia Chinen..."
7,Beautycounter Skin Twin Featherweight Foundati...,"This product bears the EWG Verified® mark, mea...",The specific formulation has been reviewed by ...,"[{'name': 'BEESWAX', 'function': 'emollient, e...","{'ingredients': '', 'directions': '', 'warning..."
8,Beautycounter Skin Twin Featherweight Foundati...,"This product bears the EWG Verified® mark, mea...",The specific formulation has been reviewed by ...,"[{'name': 'BEESWAX', 'function': 'emollient, e...","{'ingredients': '', 'directions': '', 'warning..."
9,"Well People Narcissist Foundation Sticks, Fair...","This product bears the EWG Verified® mark, mea...",The specific formulation has been reviewed by ...,"[{'name': 'ALARIA ESCULENTA EXTRACT', 'functio...","{'ingredients': 'Oleic Acid*, Ricinus Communis..."


# All Products Refined Dataset

## Description

This dataset contains information about 257 products. It includes product names, descriptions, content summaries, ingredients, and packaging information. The dataset can be used to explore and analyze product characteristics and information.

## Columns  | Data Dictionary

| Column | Description |
|---|---|
| **name** | Name of the product. |
| **description** | Detailed description of the product and its characteristics. |
| **content_summary** | Summary of the main content or features of the product. |
| **ingredients** | List or description of ingredients contained in the product. |
| **packaging** | Information about the product's packaging, container, or package type. |

## Purpose of the Dataset

This dataset can be used to explore and analyze product information, ingredients, content, and packaging characteristics. It can also be used for data cleaning, text analysis, categorization, data visualization, and other data analysis exercises.

In [96]:
df8 = '/content/DATA/obesity_rates_by_country.json'
read_json2= pd.read_json(df8)
read_json2.head(11)



,title,subtitle,updated_at,data
0,"Obesity Rates by Country (WHO, 2022)",Percentage of adults classified as obese (BMI ...,2025-12-23 10:46:00+00:00,"{'rank': 1, 'country': 'American Samoa', 'obes..."
1,"Obesity Rates by Country (WHO, 2022)",Percentage of adults classified as obese (BMI ...,2025-12-23 10:46:00+00:00,"{'rank': 2, 'country': 'Tonga', 'obesity_rate_..."
2,"Obesity Rates by Country (WHO, 2022)",Percentage of adults classified as obese (BMI ...,2025-12-23 10:46:00+00:00,"{'rank': 3, 'country': 'Nauru', 'obesity_rate_..."
3,"Obesity Rates by Country (WHO, 2022)",Percentage of adults classified as obese (BMI ...,2025-12-23 10:46:00+00:00,"{'rank': 4, 'country': 'Tokelau', 'obesity_rat..."
4,"Obesity Rates by Country (WHO, 2022)",Percentage of adults classified as obese (BMI ...,2025-12-23 10:46:00+00:00,"{'rank': 5, 'country': 'Cook Islands', 'obesit..."
5,"Obesity Rates by Country (WHO, 2022)",Percentage of adults classified as obese (BMI ...,2025-12-23 10:46:00+00:00,"{'rank': 6, 'country': 'Niue', 'obesity_rate_t..."
6,"Obesity Rates by Country (WHO, 2022)",Percentage of adults classified as obese (BMI ...,2025-12-23 10:46:00+00:00,"{'rank': 7, 'country': 'Tuvalu', 'obesity_rate..."
7,"Obesity Rates by Country (WHO, 2022)",Percentage of adults classified as obese (BMI ...,2025-12-23 10:46:00+00:00,"{'rank': 8, 'country': 'Samoa', 'obesity_rate_..."
8,"Obesity Rates by Country (WHO, 2022)",Percentage of adults classified as obese (BMI ...,2025-12-23 10:46:00+00:00,"{'rank': 9, 'country': 'French Polynesia', 'ob..."
9,"Obesity Rates by Country (WHO, 2022)",Percentage of adults classified as obese (BMI ...,2025-12-23 10:46:00+00:00,"{'rank': 10, 'country': 'Bahamas', 'obesity_ra..."


# Obesity Rates by Country Dataset

## Description

This dataset contains obesity rates for adults across 197 countries and territories based on WHO data for 2022. It includes overall obesity rates as well as separate obesity rates for males and females. Obesity is defined as a body mass index (BMI) of 30 kg/m² or higher.

## Columns  | Data Dictionary

| Column | Description |
|---|---|
| **rank** | Ranking of the country based on the total adult obesity rate. |
| **country** | Name of the country or territory. |
| **obesity_rate_total** | Percentage of adults classified as obese based on the total population. |
| **obesity_rate_male** | Percentage of adult males classified as obese. |
| **obesity_rate_female** | Percentage of adult females classified as obese. |

## Purpose of the Dataset

This dataset can be used to explore and compare adult obesity rates across countries and examine differences between male and female obesity rates. It can also be used for data visualization, statistical analysis, country-level comparisons, and identifying patterns in obesity prevalence.

In [97]:
df9 = '/content/DATA/major_epidemics_and_pandemics.json'
read_json3 = pd.read_json(df9)
read_json3.head(11)


,title,subtitle,updated_at,data
0,Major Epidemics and Pandemics,Global outbreaks with over 1 million deaths,2025-12-09 13:56:00+00:00,"{'rank': '1', 'epidemic_or_pandemic': 'Plague ..."
1,Major Epidemics and Pandemics,Global outbreaks with over 1 million deaths,2025-12-09 13:56:00+00:00,"{'rank': '2', 'epidemic_or_pandemic': 'HIV/AID..."
2,Major Epidemics and Pandemics,Global outbreaks with over 1 million deaths,2025-12-09 13:56:00+00:00,"{'rank': '3', 'epidemic_or_pandemic': 'Black D..."
3,Major Epidemics and Pandemics,Global outbreaks with over 1 million deaths,2025-12-09 13:56:00+00:00,"{'rank': '4', 'epidemic_or_pandemic': '1918 ""S..."
4,Major Epidemics and Pandemics,Global outbreaks with over 1 million deaths,2025-12-09 13:56:00+00:00,"{'rank': '5', 'epidemic_or_pandemic': 'COVID-1..."
5,Major Epidemics and Pandemics,Global outbreaks with over 1 million deaths,2025-12-09 13:56:00+00:00,"{'rank': '6', 'epidemic_or_pandemic': 'Third p..."
6,Major Epidemics and Pandemics,Global outbreaks with over 1 million deaths,2025-12-09 13:56:00+00:00,"{'rank': '7', 'epidemic_or_pandemic': 'Cocoliz..."
7,Major Epidemics and Pandemics,Global outbreaks with over 1 million deaths,2025-12-09 13:56:00+00:00,"{'rank': '8', 'epidemic_or_pandemic': 'Antonin..."
8,Major Epidemics and Pandemics,Global outbreaks with over 1 million deaths,2025-12-09 13:56:00+00:00,"{'rank': '9', 'epidemic_or_pandemic': '1520 Me..."
9,Major Epidemics and Pandemics,Global outbreaks with over 1 million deaths,2025-12-09 13:56:00+00:00,"{'rank': '10', 'epidemic_or_pandemic': '1957–1..."


## Description

This dataset contains information about 19 major epidemics and pandemics throughout history. It includes the disease name, estimated death toll, percentage of population lost, duration, and geographic location of each epidemic or pandemic.

## Columns  | Data Dictionary

| Column | Description |
|---|---|
| **rank** | Ranking of the epidemic or pandemic based on the dataset. |
| **epidemic_or_pandemic** | Name of the major epidemic or pandemic event. |
| **disease** | Disease or infectious illness associated with the event. |
| **death_toll** | Estimated number of deaths associated with the epidemic or pandemic. |
| **percentage_of_population_lost** | Estimated percentage of the affected population that died during the event. |
| **years** | Time period or years during which the epidemic or pandemic occurred. |
| **location** | Geographic region or location affected by the epidemic or pandemic. |

## Purpose of the Dataset

This dataset can be used to explore major epidemics and pandemics throughout history and compare their estimated impact, duration, and geographic spread. It can also be used for data visualization, historical analysis, statistical comparisons, and examining patterns in the impact of infectious diseases.